<?xml version="1.0"?>
<sdf version="1.7">
  <world name="my_world">
    <model name="water_plane">
      <static>true</static>
      <link name="link">
        <visual name="water">
          <geometry>
            <plane>
              <size>100 100</size>
              <normal>0 0 1</normal>
            </plane>
          </geometry>
          <material>
            <ambient>0 0 1 0.5</ambient>
            <diffuse>0 0 1 0.5</diffuse>
            <specular>0 0 1 0.5</specular>
          </material>
        </visual>
      </link>
    </model>

    <model name="red_sphere">
      <pose>2 0 0.3 0 0 0</pose>
      <static>true</static>
      <link name="link">
        <visual name="visual">
          <geometry><sphere><radius>0.2</radius></sphere></geometry>
          <material>
            <ambient>1 0 0 1</ambient>
            <diffuse>1 0 0 1</diffuse>
            <specular>1 0 0 1</specular>
          </material>
        </visual>
        <collision name="collision">
          <geometry><sphere><radius>0.2</radius></sphere></geometry>
        </collision>
      </link>
    </model>

    <include>
      <uri>model://box_usv</uri>
      <name>box_usv</name>
      <pose>0 0 0 0 0 0</pose>
    </include>

    <physics name="default_physics" type="ode">
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1.0</real_time_factor>
    </physics>

    <plugin filename="ignition-gazebo-buoyancy-system" name="ignition::gazebo::systems::Buoyancy">
      <graded_buoyancy>
        <default_density>1000</default_density>
        <density_change>
          <above_depth>0</above_depth>
          <density>1</density>
        </density_change>
      </graded_buoyancy>
    </plugin>

    <plugin filename="ignition-gazebo-physics-system" name="ignition::gazebo::systems::Physics"/>
    <plugin filename="ignition-gazebo-user-commands-system" name="ignition::gazebo::systems::UserCommands"/>
    <plugin filename="ignition-gazebo-scene-broadcaster-system" name="ignition::gazebo::systems::SceneBroadcaster"/>
  </world>
</sdf>
